core/state_transition.go

```diff
@@ -253,7 +253,8 @@ func (self *StateTransition) transitionDb() (ret []byte, usedGas *big.Int, err e
 		err = nil
 	}
 
-	if vm.Debug {
+	logs := vmenv.StructLogs()
+	if len(logs) > 0 {
```

Comment:
In order to have structured logs created you need to specify it in the config. If that's not set no structured logs will ever be emitted.

This flag has been completely removed. I'm pondering to remove this bit of code completely in favour of the new structured logging (e.g. json). Have another tool convert that to the structured log output generated by the `vm.StdErrFormat`.